app/components/collection-search.cjsx

```diff
@@ -8,40 +8,50 @@ module.exports = React.createClass
 
   getDefaultProps: ->
     multi: false
-    favorite: false
-    user: null
     project: null
 
+  getInitialState: ->
+    collections: []
+
   searchCollections: (value, callback) ->
     query =
       page_size: 10
       current_user_roles: 'owner,collaborator'
     query.search = "#{value}" unless value is ''
-    query.project_id = @props.project.id if @props.project?
 
     apiClient.type('collections').get query
       .then (collections) ->
-        opts = collections.map (collection) -> {
-          value: collection.id,
-          label: collection.display_name
-          collection: collection
-        }
+
+        opts = collections.map (collection) ->
+          {
+            value: collection.id,
+            label: collection.display_name,
+            collection: collection
+          }
 
         callback null, {
           options: opts
         }
 
-  options: ->
-    return @refs.collectionSelect.state.options
+  addSelected:(collIds) ->
```

Comment:
Yeah, i did. But it only passes the current state of the selected options so is one behind on each onChange event and would require n+1 calls to get to n. I thought there would be a nice way to just call `selectComponent.selected()` instead of hassling with `.value() manipulations, etc.`